src/main/java/frc/robot/Constants.java

```diff
@@ -4,20 +4,90 @@
 
 package frc.robot;
 
+import static frc.robot.Constants.DriveTrainConstants.DRIVE_KINEMATICS;
+import static frc.robot.Constants.DriveTrainConstants.FEED_FORWARD;
+
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.kinematics.DifferentialDriveKinematics;
+import edu.wpi.first.math.trajectory.constraint.DifferentialDriveVoltageConstraint;
+import edu.wpi.first.math.util.Units;
+
 /**
- * The Constants class provides a convenient place for teams to hold robot-wide numerical or boolean
- * constants. This class should not be used for any other purpose. All constants should be declared
+ * The Constants class provides a convenient place for teams to hold robot-wide
+ * numerical or boolean
+ * constants. This class should not be used for any other purpose. All constants
+ * should be declared
  * globally (i.e. public static). Do not put anything functional in this class.
  *
- * <p>It is advised to statically import this class (or one of its inner classes) wherever the
+ * <p>
+ * It is advised to statically import this class (or one of its inner classes)
+ * wherever the
  * constants are needed, to reduce verbosity.
  */
 public final class Constants {
-    public static final class DrivetrainConstants {
-        public static final int LEFT_LEADER_ID = 0;
-        public static final int LEFT_FOLLOWER_ID = 1;
-        public static final int RIGHT_LEADER_ID = 2;
-        public static final int RIGHT_FOLLOWER_ID = 3;
+    public static final class DriveTrainConstants {
+        public static final int DEVICE_ID_RIGHT_MASTER = 0;
+        public static final int DEVICE_ID_RIGHT_SLAVE_ONE = 1;
+        public static final int DEVICE_ID_RIGHT_SLAVE_TWO = 2;
+        public static final int DEVICE_ID_LEFT_MASTER = 3;
+        public static final int DEVICE_ID_LEFT_SLAVE_ONE = 4;
+        public static final int DEVICE_ID_LEFT_SLAVE_TWO = 5;
+
+        public static final int EDGES_PER_ROTATION = 8192;
+        public static final double WHEEL_DIAMETER_INCHES = 6d;
+        public static final double WHEEL_CIRCUMFERENCE_METERS = Units.inchesToMeters(WHEEL_DIAMETER_INCHES) * Math.PI;
+
+        public static final double TRACK_WIDTH_METERS = 0.7101550117116572;
+        public static final DifferentialDriveKinematics DRIVE_KINEMATICS = new DifferentialDriveKinematics(
+                TRACK_WIDTH_METERS);
+
+        /** Voltage needed to overcome the motor’s static friction. kS */
+        public static final double kS = 1.01;
+
+        /** Voltage needed to hold (or "cruise") at a given constant velocity. kV */
+        public static final double kV = 2.93;
+
+        /** Voltage needed to induce a given acceleration in the motor shaft. kA */
+        public static final double kA = 0.761;
+
+        public static final SimpleMotorFeedforward FEED_FORWARD = new SimpleMotorFeedforward(kS, kV, kA);
 
+        public static final double kP = 0.154;
```

Comment:
We need to run SysID and update these values. See the WPILib documentation for [System Identification](https://docs.wpilib.org/en/stable/docs/software/pathplanning/system-identification/introduction.html?highlight=characterization)
We will have to do this multiple times, whenever the weight or center of gravity on the robot changes significantly while we're adding and changing mechanisms.